HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2109,66 +2109,118 @@ def self.get_default_compressor_type(seer)
+      energy_factor, water_removal_rate = dehumidifier_ief_to_ef_inputs(w_coeff, ef_coeff, dehumidifier.integrated_energy_factor, water_removal_rate)
+    end
+
+    # Calculate air flow rate by assuming 2.75 cfm/pint/day (based on experimental test data)
+    air_flow_rate = 2.75 * water_removal_rate
+
+    humidistat = OpenStudio::Model::ZoneControlHumidistat.new(model)
+    humidistat.setName(obj_name + " #{control_zone.name} humidistat")
+    humidistat.setHumidifyingRelativeHumiditySetpointSchedule(relative_humidity_setpoint_sch)
+    humidistat.setDehumidifyingRelativeHumiditySetpointSchedule(relative_humidity_setpoint_sch)
+    control_zone.setZoneControlHumidistat(humidistat)
+
+    zone_hvac = OpenStudio::Model::ZoneHVACDehumidifierDX.new(model, water_removal_curve, energy_factor_curve, part_load_frac_curve)
+    zone_hvac.setName(obj_name + " #{control_zone.name} dx")
+    zone_hvac.setAvailabilitySchedule(model.alwaysOnDiscreteSchedule)
+    zone_hvac.setRatedWaterRemoval(UnitConversions.convert(water_removal_rate, 'pint', 'L'))
+    zone_hvac.setRatedEnergyFactor(energy_factor / dehumidifier.fraction_served)
+    zone_hvac.setRatedAirFlowRate(UnitConversions.convert(air_flow_rate, 'cfm', 'm^3/s'))
+    zone_hvac.setMinimumDryBulbTemperatureforDehumidifierOperation(10)
+    zone_hvac.setMaximumDryBulbTemperatureforDehumidifierOperation(40)
+
+    zone_hvac.addToThermalZone(control_zone)
+    # Only one dehumidifier allowed in current workflow.
+    # If more than one allowed in the future, should remove this EMS program to avoid duplication
+    hvac_map[dehumidifier.id] << zone_hvac
+    adjust_dehumidifier_load_EMS(dehumidifier.fraction_served, zone_hvac, model, living_space)
+  end
 
-      zone_hvac.addToThermalZone(control_zone)
-    end
+  def self.dehumidifier_ief_to_ef_inputs(w_coeff, ef_coeff, ief, water_removal_rate)
+    # Shift inputs under IEF test conditions to E+ supported EF test conditions
+    # test conditions
+    ief_db = UnitConversions.convert(65.0, 'F', 'C') # degree C
+    rh = 60.0 # for both EF and IEF test conditions, %
+
+    # Independent ariables applied to curve equations
+    var_array_ief = [1, ief_db, ief_db * ief_db, rh, rh * rh, ief_db * rh]
+
+    # Curved values under EF test conditions
+    curve_value_ef = 1 # Curves are nomalized to 1.0 under EF test conditions, 80F, 60%
+    # Curve values under IEF test conditions
+    ef_curve_value_ief = var_array_ief.zip(ef_coeff).map { |var, coeff| var * coeff }.inject(0, :+)
+    water_removal_curve_value_ief = var_array_ief.zip(w_coeff).map { |var, coeff| var * coeff }.inject(0, :+)
+
+    # E+ inputs under EF test conditions
+    ef_input = ief / ef_curve_value_ief * curve_value_ef
+    water_removal_rate_input = water_removal_rate / water_removal_curve_value_ief * curve_value_ef
+
+    return ef_input, water_removal_rate_input
+  end
+
+  def self.adjust_dehumidifier_load_EMS(fraction_served, zone_hvac, model, living_space)
```

Comment:
Please add comments describing why this EMS program is needed and what it does.

Reply:
Also, what do we think about skipping EMS when the dehumidifier serves 100% of the load?